### PR TITLE
Use new configuration depreciation method

### PR DESCRIPTION
**Pull request by sanpii**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

In keeping with #22382

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php [hunk 1]
```diff
@@ -64,8 +64,6 @@ public function boot()
         ErrorHandler::register(null, false)->throwAt($this->container->getParameter('debug.error_handler.throw_at'), true);
 
         if ($this->container->hasParameter('kernel.trusted_proxies')) {
-            @trigger_error('The "kernel.trusted_proxies" parameter is deprecated since version 3.3 and will be removed in 4.0. Use the Request::setTrustedProxies() method in your front controller instead.', E_USER_DEPRECATED);
```
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: this change should be reverted as the notice is not about the `framework.trusted_proxies` config option but the corresponding kernel parameter, triggering is still relevant. If you think it should be rediscussed, that should be done on 3.3 where it has been introduced.

**Review by chalasr — CHANGES_REQUESTED** [review]

**Comment by sanpii** [comment]
I don’t understand why the tests failed on travis. I can’t reproduce the problem on my computer (with php 7.0 and `--prefer-lowest`).

**Comment by chalasr** [comment]
Should be better after #24083.

**Comment by nicolas-grekas** [comment]
rebase needed to account for #24083.

**Comment by sanpii** [comment]
Done, thank you @chalasr.

It’s better but it still failed with hhvm and php 5.5 due to deprecation notices.

**Comment by chalasr** [comment]
@sanpii Looks like this triggers extra deprecations, I suspect that we trigger when we should not (e.g. because the node has a default value, we trigger even if not explicitly set). I'll look into later today.

**Review by chalasr — DISMISSED** [review]
That was the case (https://github.com/symfony/symfony/pull/24025#issuecomment-326909876), added a commit that makes nodes trigger deprecations only if the value is explicitly set. PR looks good to me.

**Comment by chalasr** [comment]
VariableNode (= all nodes which are not `ArrayNode`) should not trigger the deprecation itself, as it is always the child of a rootNode which is responsible for triggering when a child is marked as deprecated. PR ready (fixing the implementation)

**Comment by nicolas-grekas** [comment]
Thank you @sanpii and @chalasr

**Pull request merged**